src/main/java/com/incognia/IncogniaAPI.java

```diff
@@ -83,32 +84,31 @@ public IncogniaAPI(String clientId, String clientSecret, Region region) {
    *                     .build())
    *             .coordinates(new Coordinates(40.74836007062138, -73.98509720487937))
    *             .build();
-   *      SignupAssessment assessment = api.registerSignup("installation id", address);
+   *      RegisterSignupRequest signupRequest = RegisterSignupRequest.builder().installationId(installationId).address(address).build();
+   *      SignupAssessment assessment = api.registerSignup(signupRequest);
    * } catch (IncogniaAPIException e) {
    *      //Some api error happened (invalid data, invalid credentials)
    * } catch (IncogniaException e) {
    *      //Something unexpected happened
    * }
    * }</pre>
    *
-   * @param installationId the <a
-   *     href="https://docs.incognia.com/learning/concepts/identifiers/installation-id">installation
-   *     id</a>
-   * @param address the address
+   * @param request the {@link RegisterSignupRequest} model that contains the properties we need to
+   *     make an assessment.
    * @return the assessment
    * @throws IncogniaAPIException in case of api errors
    * @throws IncogniaException in case of unexpected errors
    */
-  public SignupAssessment registerSignup(String installationId, Address address)
-      throws IncogniaException {
-    Asserts.assertNotEmpty(installationId, "installation id");
-    Asserts.assertNotNull(address, "address");
+  public SignupAssessment registerSignup(RegisterSignupRequest request) throws IncogniaException {
```

Comment:
Shouldn't we support the old method overloaded as well, to avoid breaking changes? 
(Same applies for other methods)

Reply:
We decided to go on with these breaking changes and go up one major version